Documentation/design-docs/AssemblyLoadContext.Activate.md

```diff
@@ -0,0 +1,306 @@
+}
+
+void SomeMethod()
+{
+  using (AssemblyLoadContext.Activate(null))
+  {
+    // AssemblyLoadContext.ActiveContext is null
+    // Behavior prior to .NET Core 3.0 is unchanged
+    Assembly assembly = Assembly.Load(myPlugin); // Will load into the ALC containing SomeMethod().
+  }
+}
+```
+
+### Advanced/Problematic use cases
+When initially running code within an `AssemblyLoadContext` it is relatively easy to correctly set the `ActiveContext`. It is more difficult to determine the other places where code is or isn't crossing the intended `AssemblyLoadContext` boundary. This is likely to include:
+* Event Callback Handlers: user interface, disk, timers, network ...
+* Callbacks
+
+The following are representative patterns that will need to be supported for these more complicated usages.
+
+#### An incoming event handler into an AssemblyLoadContext
+```C#
+void OnEvent()
+{
+  using (alc.Activate())
+  {
+    ...
+  }
+}
+```
+#### An incoming event handler into a collectible AssemblyLoadContext
+```C#
+class WeakAssemblyLoadContextEventHandler
+{
+  WeakReference<AssemblyLoadContext> weakAlc;
+
+  void OnEvent()
+  {
+    AssemblyLoadContext alc;
+    if(weakAlc.TryGetTarget(out alc))
+    {
+      using (alc.Activate())
+      {
+        ...
+      }
+    }
+  }
+}
+```
+#### A outgoing callback from an ActiveContext
+```C#
+using (AssemblyLoadContext.Activate(null))
+{
+  Callback();
+}
+```
+#### An outgoing event handler from an ActiveContext
+```C#
+void OnEvent()
+{
+  using (AssemblyLoadContext.Activate(null))
+  {
+    ...
+  }
+}
+```
+### Unsupported - Wish list
+Events, interfaces and reflection are main mechanisms for loosely coupling components. It would be nice for them to just work with `ActiveContext`.  For instance:
+#### Automatic event handler wrapping
+```C#
+using (alc.Activate())
+{
+  // Call OnTimedEvent with alc as ActiveContext
+  aTimer.Elapsed += OnTimedEvent;
+}
+```
+#### Interface proxies
+```C#
+using (alc)
+{
+  MyOutgoingInterface out = InterfaceDeActivateContext<MyOutgoingInterface>(outRaw);
+  MyIncomingInterface in = InterfaceActivateContext<MyIncomingInterface>(inRaw);
+}
+```
+## Motivational Background
+### .NET Framework Behavior
+
+There are a set of APIs whose behavior is dependent on the current AppDomain.
+
+These APIs implicitly affect and are affected by the current AppDomain. Generally these are APIs which load assemblies or resolve types from strings.
+
+Typical representative APIs would be:
+
++ `Assembly.Load(AssemblyName)`
++ `Assembly.GetType(String)`
+
+This document will refer to the complete set of the APIS as `context sensitive APIs`
+### .NET Core <= 2.x Behavior
+
+The set of APIS which were depending on AppDomain, were mapped to being dependent on the AssemblyLoadContext containing the current executing assembly.
+
+This is typically implemented using a stack crawl mark to look for the assembly containing the caller of the `context sensitive API`.
+
+### Issue
```

Comment:
Please move the issue to the top of the document. I want to know what problem I am reading about prior to hearing the solution.